devo/sender/data.py

```diff
@@ -625,3 +644,155 @@ def emit(self, record):
                       severity=severity)
         except Exception:
             self.handleError(record)
+
+    @staticmethod
```

Comment:
**suggestion:** As you might have noticed, all the certificates' validation methods are static. Since this file is ever-growing (almost 800 lines now), I would suggest you convert these methods into plain functions (or move them to a separate class) and move them (or the class) to a different file.